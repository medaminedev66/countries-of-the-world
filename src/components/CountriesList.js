import React, { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Redirect } from 'react-router-dom';
import Country from './Country';
import { selectCountry } from '../redux/countries/countries';
import RandomCountry from './RandomCountry';

function CountriesList() {
  const countries = useSelector((state) => state.countriesReducer);
  const settings = useSelector((state) => state.settingReducer);
  const [state, redirect] = useState({ redirect: false });
  const dispatch = useDispatch();

  const setRedirect = (name) => {
    redirect({
      redirect: true,
    });
    dispatch(selectCountry(name));
  };

  const renderRedirect = () => {
    if (state.redirect) {
      return <Redirect to="/react-capstone/details" />;
    }
    return '';
  };

  const compare = (a, b) => {
    const type = settings[0].selectBy;
    if (a[type] < b[type]) {
      return 1;
    }
    if (a[type] > b[type]) {
      return -1;
    }
    return 0;
  };

  return (
    <div>
      <div className="continent">
        <RandomCountry countries={countries} />
      </div>
      <div className="countries-title">All countries Selected by Alphabets</div>
      <div className="countries">
        {renderRedirect()}
        {countries.sort(compare).map((country) => (
          <div
            key={country.name}
            className="countr"
            onClick={() => setRedirect(country.name)}
            aria-hidden="true"
          >
            <Country obj={country} />
          </div>
        ))}
      </div>
    </div>
  );
}

export default CountriesList;
