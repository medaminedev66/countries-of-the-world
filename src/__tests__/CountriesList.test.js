import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import CountriesList from '../components/CountriesList';
import store from '../redux/configureStore';

describe('test the CountriesList component', () => {
  it('matches the snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <CountriesList />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly', () => {
    const { queryByTestId } = render(
      <Provider store={store}>
        <CountriesList />
      </Provider>,
    );
    expect(queryByTestId('countries')).toBeTruthy();
  });
});
