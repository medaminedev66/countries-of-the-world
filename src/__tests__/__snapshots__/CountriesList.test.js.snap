// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test the CountriesList component matches the snapshot 1`] = `
<div>
  <div
    className="continent"
  >
    <div
      aria-hidden="true"
      className="Random-country"
      data-testid="Random-country"
      onClick={[Function]}
    >
      <div
        className="country-select"
      >
        <img
          alt="flag"
          className="flag"
          src="loading.png"
        />
        <h2>
          Picking...
        </h2>
      </div>
      <p>
        Let's discoverÂ 
        <span>
          ?
        </span>
      </p>
    </div>
  </div>
  <div
    className="countries-title"
  >
    All countries Selected by Alphabets
  </div>
  <div
    className="countries"
    data-testid="countries"
  >
    
  </div>
</div>
`;
